src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -40,19 +50,55 @@
 public class DomainApis {
 
     private static final Logger logger = Logger.getLogger(DomainApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+
+    /**
+     * init the DB
+     *
+     *
+     */
+    public void init(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+    }
 
     /**
      * Creates the Policy for a given domain
      *
      * @return policy
      */
-    @Path("/{domainid}")
+    @Path("/{domainid}/policies")
     @POST
     public PolicyEntity createPolicy(@PathParam("domainid") Long domainid, PolicyEntity entity) {
 
-        logger.info("Creating Policy Entity...");
-        //TODO Sudhir: Add db calls here
-        return null;
+        logger.info("Creating Policy Entity...:" + entity.getName());
+
+        return this.txControl.required(new Callable<PolicyEntity>() {
+
+            @Override
+            public PolicyEntity call() throws Exception {
+
+                CriteriaBuilder cb = em.getCriteriaBuilder();
+
+                CriteriaQuery<DomainEntity> q = cb.createQuery(DomainEntity.class);
+                Root<DomainEntity> r = q.from(DomainEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("Id"), domainid)));
+
+                List<DomainEntity> result = em.createQuery(q).getResultList();
+                if (result.isEmpty() == false) {
+                    PolicyEntity policy = new PolicyEntity();
+                    policy.setName(entity.getName());
+                    policy.setDomain(result.get(0));
+                    policy.setParent(result.get(0).getapplianceManagerConnector());
+                    em.persist(policy);
+                    return entity;
+                }
+                else {
```

Comment:
if a domain is not found this API must return 404, not found, rather than failing silently

Reply:
The current framework does not support throwing exception 404. I will create a user story for the same and handle the exception. I will add a TODO note here.

Reply:
please don return no_domain_id as a workaround. I dont think you need support from the framework for this... basically what you need is: if the domain is not found or null log the situation and throw an exception. This will likely return a 500 in the rest api (please confirm using fiddler) but that is better than returning a "Strange" id, at least we are being explicit. Later you can handle this exception and return a 404 instead that you can follow up later.